core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -114,6 +116,29 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
   }
 
   /**
+    * Find any logs that are compact_and_delete that have segments
+    * ready for deletion
+    */
+  def logsReadyToBeTruncated(): Iterable[(TopicAndPartition, Log)] = {
```

Comment:
logsReadyToBeTruncated can be a bit confusing since truncation can be from the head or the tail. How about logsWithSegmentsToBeDeleted or sth like that?